### PR TITLE
Verison's language related attributes

### PR DESCRIPTION
**Pull request by YishaiGlasner**
rearrange the language related attributes of Version:
1. change 'actualLanguage', 'languageFamilyName', 'isSource', 'isPrimary' and 'direction' to be required.
2. change _normalize to set default values for those attributes before saving.
3. remove the attribute 'isBaeText'.

### CODE REVIEW

File: docs/openAPI.json [hunk 1]
```diff
@@ -6697,7 +6661,6 @@
               "purchaseInformationURL": "",
               "shortVersionTitle": "",
               "shortVersionTitleInHebrew": "",
```
Comment (saengel) [inline docs/openAPI.json:6663]: I think the last property cannot have a comma following it. 

i.e. all cases like this need to shift from (current)

```
 "shortVersionTitle": "",
 "shortVersionTitleInHebrew": "",
}
```

to 

```
 "shortVersionTitle": "",
 "shortVersionTitleInHebrew": ""
}
```

Do you see where the comma was deleted? I know this is super nitpicky, my apologies! Unfortunately, it's not passing validation this way...

File: docs/openAPI.json [hunk 2]
```diff
@@ -6717,7 +6680,6 @@
...
-              "isBaseText": true
```
Comment (saengel) [inline docs/openAPI.json:6720]: Same comma issue here

File: docs/openAPI.json [hunk 3]
```diff
@@ -6737,7 +6699,6 @@
               "purchaseInformationURL": "",
               "shortVersionTitle": "Dr. Joshua Kulp",
               "shortVersionTitleInHebrew": "",
-              "isBaseText": false
```
Comment (saengel) [inline docs/openAPI.json:6740]: Same comma issue here

File: docs/openAPI.json [hunk 4]
```diff
@@ -6757,7 +6718,6 @@
               "purchaseInformationURL": "",
               "shortVersionTitle": "",
               "shortVersionTitleInHebrew": "",
-              "isBaseText": false
```
Comment (saengel) [inline docs/openAPI.json:6760]: Same comma issue here

File: docs/openAPI.json [hunk 5]
```diff
@@ -6777,7 +6737,6 @@
               "purchaseInformationURL": "",
               "shortVersionTitle": "Lazarus Goldschmidt, 1929 ",
               "shortVersionTitleInHebrew": "",
-              "isBaseText": false
```
Comment (saengel) [inline docs/openAPI.json:6780]: Same comma issue here

File: docs/openAPI.json [hunk 6]
```diff
@@ -6797,7 +6756,6 @@
               "purchaseInformationURL": "",
               "shortVersionTitle": "",
               "shortVersionTitleInHebrew": "",
-              "isBaseText": false
```
Comment (saengel) [inline docs/openAPI.json:6800]: Same comma issue here

File: docs/openAPI.json [hunk 7]
```diff
@@ -6817,7 +6775,6 @@
...
```
Comment (saengel) [inline docs/openAPI.json:6820]: Same comma issue here

File: docs/openAPI.json [hunk 8]
```diff
@@ -6837,7 +6794,6 @@
               "purchaseInformationURL": "",
               "shortVersionTitle": "",
               "shortVersionTitleInHebrew": "",
-              "isBaseText": ""
```
Comment (saengel) [inline docs/openAPI.json:6840]: Same comma issue here

**Review by saengel — DISMISSED** [review]
The OpenAPI schema overall looks good - just a comma issue causing it to still fail validation. Let me know once it's fixed, and I'll test it again. Thanks!

File: sefaria/model/text.py [hunk 9]
```diff
@@ -1375,14 +1366,28 @@ def _check_node_offsets(self, content, node):
...
+            except KeyError:
+                self.languageFamilyName = constants.LANGUAGE_CODES[self.language]
+
+        self.isSource = getattr(self, "isSource", False)
+        if not getattr(self, "isPrimary", False):
+            if self.isSource or not VersionSet({'title': self.title}):
+                self.isPrimary = True
             else:
-                self.actualLanguage = self.language
+                self.isPrimary = False
```
Comment (akiva10b) [inline sefaria/model/text.py:1390]: would something like this work here?
`self.isPrimary = not getattr(self, "isPrimary", False) and (self.isSource or not VersionSet({'title': self.title}))`

File: sefaria/model/text.py [hunk 10]
```diff
@@ -1375,16 +1366,22 @@ def _check_node_offsets(self, content, node):
...
             if languageCode and languageCode.group(1):
-                self.actualLanguage = languageCode.group(1)
-            else:
-                self.actualLanguage = self.language
-
-        if not getattr(self, 'direction', None):
+                actualLanguage = languageCode.group(1)
+        if not actualLanguage:
+            actualLanguage = self.language
+        self.actualLanguage = actualLanguage
```
Comment (akiva10b) [inline sefaria/model/text.py:1377]: can be one line
`self.actualLanguage = actualLanguage or self.language`

**Comment by saengel** [comment]
The fix passed validation, all is working now from the docs perspective. Thanks @YishaiGlasner!

File: sefaria/model/tests/text_test.py [hunk 11]
```diff
@@ -743,53 +743,62 @@ def setup_class(cls):
...
                 c.delete()
             except Exception:
                 pass
     
-    def test_normalizes_actualLanguage_from_brackets(self):
-        assert self.versionWithTranslation.actualLanguage == "fr"
-    
-    def test_normalizes_language_from_language(self):
-        assert self.versionWithoutTranslation.actualLanguage == "he"
+    def test_normalize(self):
```
Comment (nsantacruz) [inline sefaria/model/tests/text_test.py:785]: It would be better to use pytest.parametrize so this test isn't so repetitive

**Review by nsantacruz — APPROVED** [review]

**Pull request merged**